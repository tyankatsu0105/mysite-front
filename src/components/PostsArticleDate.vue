<template>
	<div class="PostsArticleDate">
		<FontAwesomeIcon
			:icon="{ prefix: 'fa', iconName: 'clock' }"
		/>
		<time
			:datetime="time"
			class="PostsArticleDate-DateText"
		>
			{{ text }}
		</time>
	</div>
</template>

<script>
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
import { library } from "@fortawesome/fontawesome-svg-core";
import { faClock } from "@fortawesome/free-solid-svg-icons";
export default {
  components: {
    FontAwesomeIcon
  },
  props: {
    text: {
      type: String,
      requirer: true,
      default: ""
    },
    time: {
      type: String,
      requirer: false,
      default: ""
    }
  },
  created() {
    library.add(faClock);
  }
};
</script>

<style scoped lang='scss'>
.PostsArticleDate {
  &-DateText {
    margin-left: 5px;
  }
}
</style>
