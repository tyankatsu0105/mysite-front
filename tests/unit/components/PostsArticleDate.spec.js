import { shallowMount } from "@vue/test-utils";
import PostsArticleDate from "@/components/PostsArticleDate.vue";

const propsData = {
  time: `2018.12.15`,
  text: `2018.12.15`
};

describe("PostsArticleDate", () => {
  it("render", () => {
    const wrapper = shallowMount(PostsArticleDate, {
      propsData,
      stubs: ["font-awesome-icon"]
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
