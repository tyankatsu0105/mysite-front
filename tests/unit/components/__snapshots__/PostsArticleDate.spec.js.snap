// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsArticleDate render 1`] = `
<div class="PostsArticleDate">
  <font-awesome-icon-stub icon="[object Object]"></font-awesome-icon-stub> <time datetime="2018.12.15" class="PostsArticleDate-DateText">
    2018.12.15
  </time>
</div>
`;
